common/scala/src/main/scala/whisk/core/containerpool/logging/SplunkLogStore.scala

```diff
@@ -0,0 +1,167 @@
+    extends LogDriverLogStore(actorSystem, config) {
+  implicit val as = actorSystem
+  implicit val ec = as.dispatcher
+  implicit val materializer = ActorMaterializer()
+
+  private val splunkApi = "/services/search/jobs" //see http://docs.splunk.com/Documentation/Splunk/6.6.3/RESTREF/RESTsearch#search.2Fjobs
+
+  val log = actorSystem.log
+  val maxPendingRequests = 500
+
+  val defaultHttpFlow = Http().cachedHostConnectionPoolHttps[Promise[HttpResponse]](
+    host = config.host,
+    port = config.port,
+    connectionContext =
+      if (config.disableSNI)
+        Http().createClientHttpsContext(AkkaSSLConfig().mapSettings(s => s.withLoose(s.loose.withDisableSNI(true))))
+      else Http().defaultClientHttpsContext)
+
+  override def fetchLogs(activation: WhiskActivation): Future[ActivationLogs] = {
+
+    //example curl request:
+    //    curl -u  username:password -k https://splunkhost:port/services/search/jobs -d exec_mode=oneshot -d output_mode=json -d "search=search index=\"someindex\" | spath=activation_id | search activation_id=a930e5ae4ad4455c8f2505d665aad282 |  table log_message" -d "earliest_time=2017-08-29T12:00:00" -d "latest_time=2017-10-29T12:00:00"
+    //example response:
+    //    {"preview":false,"init_offset":0,"messages":[],"fields":[{"name":"log_message"}],"results":[{"log_message":"some log message"}], "highlighted":{}}
+    val search =
+      s"""search index="${config.index}"| spath ${config.activationIdField} | search ${config.activationIdField}=${activation.activationId.toString} | table ${config.logMessageField}"""
+
+    val formatter = DateTimeFormatter.ofPattern("YYYY-MM-dd'T'hh:mm:ss").withZone(ZoneId.of("UTC"))
+    val entity = FormData(
+      Map(
+        "exec_mode" -> "oneshot",
+        "search" -> search,
+        "output_mode" -> "json",
+        "earliest_time" -> formatter.format(activation.start),
+        "latest_time" -> formatter.format(activation.end))).toEntity
+
+    log.debug("sending request")
+    queueRequest(
+      Post(splunkApi)
+        .withEntity(entity)
+        .withHeaders(List(Authorization(BasicHttpCredentials(config.username, config.password)))))
+      .flatMap(response => {
+        log.debug(s"splunk API response ${response}")
+        Unmarshal(response.entity)
+          .to[String]
+          .map(resultsString => {
+            if (response.status.isSuccess()) {
+              log.debug(s"splunk API results: ${resultsString}")
+              val jsObject = JsonParser(resultsString).asJsObject
+              //format of results is detailed here: http://docs.splunk.com/Documentation/Splunk/latest/RESTUM/RESTusing#Example_B:_JSON_response_format_example
+              val messages = jsObject
+                .fields("results")
+                .convertTo[JsArray]
+                .elements
+                .map(msgJsValue => {
+                  msgJsValue.asJsObject.fields(config.logMessageField).asInstanceOf[JsString].value
+                })
```

Comment:
Try to never use `asInstanceOf`, there's usually a way around it. In this case:

```scala
resultsString.parseJson.convertTo[Seq[JsObject]].map(_.fields(config.logMessageField).convertTo[String])
```

Or to dense this down even more:

Use `Unmarshal` to do the heavy lifting for you, like so:

```scala
Unmarshal(response.entity).to[Seq[JsObject]].map(_.fields(config.logMessageField).convertTo[String])
```